source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -308,19 +312,18 @@ namespace aspect
 
             for (unsigned int grain_i = 0; grain_i < n_grains; ++grain_i)
               {
-                ref_volume_fractions_grains(data_position,data,mineral_i,grain_i) *= inv_sum_volume_mineral;
-                Assert(isfinite(ref_volume_fractions_grains(data_position,data,mineral_i,grain_i)),
+                set_volume_fractions_grains(data_position,data,mineral_i,grain_i,get_volume_fractions_grains(data_position,data,mineral_i,grain_i*inv_sum_volume_mineral));
```

Comment:
I think you misplaced a parenthesis in the last call. Make sure this is identical to the earlier version. I would recommend:
```
const double volume_fraction = get_volume_fractions_grains(data_position,data,mineral_i,grain_i) * inv_sum_volume_mineral;
set_volume_fractions_grains(...)
```